### PR TITLE
getDisplayField return string

### PR DESCRIPTION
**Pull request by ava007**
#544

### CODE REVIEW

File: src/Command/ModelCommand.php [hunk 1]
```diff
@@ -483,7 +483,7 @@ public function getDisplayField(\Cake\ORM\Table $model, \Cake\Console\Arguments
             return $args->getOption('display-field');
         }
 
-        return $model->getDisplayField();
+        return (string)$model->getDisplayField();
```
Comment (ADmad) [inline src/Command/ModelCommand.php]: Might be better to change the method's return type hint to allow returning `null` instead. So `: string` to `: ?string`.

File: src/Command/ModelCommand.php [hunk 2]
```diff
@@ -477,7 +477,7 @@ public function findBelongsToMany(\Cake\ORM\Table $model, array $associations):
      * @param \Cake\Console\Arguments $args CLI Arguments
      * @return string
```
Comment (ADmad) [inline src/Command/ModelCommand.php]: This should now be `@return string|null`

**Comment by saeideng** [comment]
@ADmad `getDisplayField()` can returns array , true ??
https://github.com/cakephp/cakephp/blob/cc12b275c6775c4cf70d30f966dfc66746cef249/src/ORM/Table.php#L613-L615

**Comment by dereuromark** [comment]
@saeideng No, I think that is a wrongly declared array typehint here. only scalar allowed.
fd9f3931109ffcc03c6d49084cebbd6277849e3f introduced that as part of some cleanup, but i am sure this wasnt correct. Even for multi-primary-key cases only one (the first) qualifies.

**Comment by ADmad** [comment]
I'll check the core and confirm.

**Comment by dereuromark** [comment]
Any update on this one?

File: src/Command/ModelCommand.php [hunk 3]
```diff
@@ -475,9 +475,9 @@ public function findBelongsToMany(\Cake\ORM\Table $model, array $associations):
      *
      * @param \Cake\ORM\Table $model The model to introspect.
      * @param \Cake\Console\Arguments $args CLI Arguments
-     * @return string
+     * @return string|null
      */
-    public function getDisplayField(\Cake\ORM\Table $model, \Cake\Console\Arguments $args): string
+    public function getDisplayField(\Cake\ORM\Table $model, \Cake\Console\Arguments $args): ?string
```
Comment (ADmad) [inline src/Command/ModelCommand.php]: Return value for core's `Table:getDisplayField()` is `string|string[]` so adding return typehints of `string` or `?string` both would be incorrect here.

Though we should ensure that wherever the return values of this method is consumed handles array values properly too.

**Pull request merged**